quic/s2n-quic-core/src/crypto/tls.rs

```diff
@@ -53,6 +51,11 @@ pub trait Context<Crypto: CryptoSuite> {
         application_parameters: ApplicationParameters,
     ) -> Result<(), transport::Error>;
 
+    fn on_server_name(
+        &mut self,
+        server_name: Option<crate::application::ServerName>,
```

Comment:
This should only be called if there is a server name present so I would get rid of the Option